restio/fields/base.py

```diff
@@ -10,6 +19,11 @@
 T_co = TypeVar("T_co", bound=object, covariant=True)
 SubT = TypeVar("SubT")
 
+# stores the names for a model class type
+# this object stays here and is globaly declared so that it can be
+# accessed by both BaseModelMeta and Field instances
+MODEL_TYPE_REGISTRY: Dict[str, Type] = {}
```

Comment:
Since it seems it is only necessary (except in the tests) to add elements to this registry and read from it, maybe it should be managed by a singleton. This way access to certain functions (like remove or clear) can be controlled.

Reply:
I thought about this, and decided to make it very simple because I don't foresee any further uses for it. I will leave a comment to refactor in case we need to use if for anything else